### PR TITLE
wrong exception is sent due to separator being hard coded to "?" when raising an error

### LINKED ISSUE #238: redirect uri on error (e.g. AuthCanceled) may contain two question marks, thus sending the wrong exception
I'm using django-oauth-toolkit along with python-social-auth and of course oauthlib which is a requirement of both, and I've run into an issue when the user clicks the "cancel" button. 

The wrong exception may occur because of a redirect problem.  When the user clicks "cancel", if the redirect_uri contains any URL parameters, an incorrect state error is given instead of `AuthCanceled`, because the URI ends up something like:

`http://example.com?redirect_state=SOME_STATE_TOKEN?error=access_denied`

Note the two question marks here. This is due to an issue in mixins.py - a pull request that fixes this is forthcoming.

### PR DESCRIPTION
**Pull request by honestbleeps**
This checks for the presence of a question mark in the URI and changes the separator accordingly so as to not put two questions marks in the URI.

fixes #238

### CODE REVIEW

**Comment by masci** [comment]
Changes look good to me. Do you mind to fix the tests?

**Comment by honestbleeps** [comment]
sorry about that, I will check out the tests today!

**Comment by honestbleeps** [comment]
tests are running now, should be fixed!

File: oauth2_provider/views/mixins.py [hunk 1]
```diff
@@ -155,9 +155,17 @@ def error_response(self, error, **kwargs):
         :param error: :attr:`OAuthToolkitError`
         """
         oauthlib_error = error.oauthlib_error
+
+        separator = '?'
+        try:
+            if '?' in oauthlib_error.redirect_uri:
+                separator = '&'
+        except:
+            pass
```
Comment (masci) [inline oauth2_provider/views/mixins.py]: sorry for being pedantic but I don't like `except: pass` very much ;)
What about something like:
```
redirect_uri = oauthlib_error.redirect_uri or ""
separator = '&' if '?' in redirect_uri else '?'
```
Also, I guess a test case checking the double ? problem (a test that fails without your patch) should be added

**Pull request merged**